diff --git a/ruoyi-common/src/main/java/com/ruoyi/common/core/domain/entity/SysMenu.java b/ruoyi-common/src/main/java/com/ruoyi/common/core/domain/entity/SysMenu.java
index 71035128..9680c511 100644
--- a/ruoyi-common/src/main/java/com/ruoyi/common/core/domain/entity/SysMenu.java
+++ b/ruoyi-common/src/main/java/com/ruoyi/common/core/domain/entity/SysMenu.java
@@ -183,12 +183,12 @@ public class SysMenu extends BaseEntity
 
     public List<SysMenu> getChildren()
     {
-        return children;
+        return new ArrayList<SysMenu>(children);
     }
 
     public void setChildren(List<SysMenu> children)
     {
-        this.children = children;
+        this.children = children == null ? new ArrayList<SysMenu>() : new ArrayList<SysMenu>(children);
     }
 
     @Override
